atmo/jobs/jobs.py

```diff
@@ -13,9 +13,9 @@ def run_jobs():
     """
     run_jobs = []
     for job in SparkJob.objects.all():
-        # first let's update the status to update most_recent_status
-        job.update_status()
-        job.save()
+        # first let's update the status if there is a prior run
+        if job.latest_run:
```

Comment:
I can't think of anything and being a simple query won't it be pretty quick? If we put an index on the date column that would help once there are lots of runs.

Otherwise perhaps go back to adding the denormalized `last_run_id` column on `SparkJob` so we can query them with a JOIN in a single query?